# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-region.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-region.c
--- a/gfx/cairo/cairo/src/cairo-region.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-region.c	Mon May 06 19:25:12 2024 +0000
@@ -111,6 +111,8 @@
     case CAIRO_STATUS_FREETYPE_ERROR:
     case CAIRO_STATUS_WIN32_GDI_ERROR:
     case CAIRO_STATUS_TAG_ERROR:
+    case CAIRO_STATUS_DWRITE_ERROR:
+    case CAIRO_STATUS_SVG_FONT_ERROR:
     default:
 	_cairo_error_throw (CAIRO_STATUS_NO_MEMORY);
 	return (cairo_region_t *) &_cairo_region_nil;
@@ -211,7 +213,6 @@
 
     return region;
 }
-slim_hidden_def (cairo_region_create);
 
 /**
  * cairo_region_create_rectangles:
@@ -279,7 +280,6 @@
 
     return region;
 }
-slim_hidden_def (cairo_region_create_rectangles);
 
 cairo_region_t *
 _cairo_region_create_from_boxes (const cairo_box_t *boxes, int count)
@@ -345,7 +345,6 @@
 
     return region;
 }
-slim_hidden_def (cairo_region_create_rectangle);
 
 /**
  * cairo_region_copy:
@@ -382,7 +381,6 @@
 
     return copy;
 }
-slim_hidden_def (cairo_region_copy);
 
 /**
  * cairo_region_reference:
@@ -407,7 +405,6 @@
     _cairo_reference_count_inc (&region->ref_count);
     return region;
 }
-slim_hidden_def (cairo_region_reference);
 
 /**
  * cairo_region_destroy:
@@ -433,7 +430,6 @@
     _cairo_region_fini (region);
     free (region);
 }
-slim_hidden_def (cairo_region_destroy);
 
 /**
  * cairo_region_num_rectangles:
@@ -453,7 +449,6 @@
 
     return pixman_region32_n_rects (CONST_CAST &region->rgn);
 }
-slim_hidden_def (cairo_region_num_rectangles);
 
 /**
  * cairo_region_get_rectangle:
@@ -485,7 +480,6 @@
     rectangle->width = pbox->x2 - pbox->x1;
     rectangle->height = pbox->y2 - pbox->y1;
 }
-slim_hidden_def (cairo_region_get_rectangle);
 
 /**
  * cairo_region_get_extents:
@@ -515,7 +509,6 @@
     extents->width = pextents->x2 - pextents->x1;
     extents->height = pextents->y2 - pextents->y1;
 }
-slim_hidden_def (cairo_region_get_extents);
 
 /**
  * cairo_region_status:
@@ -533,7 +526,6 @@
 {
     return region->status;
 }
-slim_hidden_def (cairo_region_status);
 
 /**
  * cairo_region_subtract:
@@ -564,7 +556,6 @@
 
     return CAIRO_STATUS_SUCCESS;
 }
-slim_hidden_def (cairo_region_subtract);
 
 /**
  * cairo_region_subtract_rectangle:
@@ -598,7 +589,6 @@
 
     return status;
 }
-slim_hidden_def (cairo_region_subtract_rectangle);
 
 /**
  * cairo_region_intersect:
@@ -625,7 +615,6 @@
 
     return CAIRO_STATUS_SUCCESS;
 }
-slim_hidden_def (cairo_region_intersect);
 
 /**
  * cairo_region_intersect_rectangle:
@@ -660,7 +649,6 @@
 
     return status;
 }
-slim_hidden_def (cairo_region_intersect_rectangle);
 
 /**
  * cairo_region_union:
@@ -688,7 +676,6 @@
 
     return CAIRO_STATUS_SUCCESS;
 }
-slim_hidden_def (cairo_region_union);
 
 /**
  * cairo_region_union_rectangle:
@@ -722,7 +709,6 @@
 
     return status;
 }
-slim_hidden_def (cairo_region_union_rectangle);
 
 /**
  * cairo_region_xor:
@@ -761,7 +747,6 @@
 
     return status;
 }
-slim_hidden_def (cairo_region_xor);
 
 /**
  * cairo_region_xor_rectangle:
@@ -802,7 +787,6 @@
 
     return status;
 }
-slim_hidden_def (cairo_region_xor_rectangle);
 
 /**
  * cairo_region_is_empty:
@@ -822,7 +806,6 @@
 
     return ! pixman_region32_not_empty (CONST_CAST &region->rgn);
 }
-slim_hidden_def (cairo_region_is_empty);
 
 /**
  * cairo_region_translate:
@@ -843,7 +826,6 @@
 
     pixman_region32_translate (&region->rgn, dx, dy);
 }
-slim_hidden_def (cairo_region_translate);
 
 /**
  * cairo_region_contains_rectangle:
@@ -884,7 +866,6 @@
     case PIXMAN_REGION_PART: return CAIRO_REGION_OVERLAP_PART;
     }
 }
-slim_hidden_def (cairo_region_contains_rectangle);
 
 /**
  * cairo_region_contains_point:
@@ -909,7 +890,6 @@
 
     return pixman_region32_contains_point (CONST_CAST &region->rgn, x, y, &box);
 }
-slim_hidden_def (cairo_region_contains_point);
 
 /**
  * cairo_region_equal:
@@ -940,4 +920,3 @@
 
     return pixman_region32_equal (CONST_CAST &a->rgn, CONST_CAST &b->rgn);
 }
-slim_hidden_def (cairo_region_equal);